test/soldiers.test.js

```diff
@@ -0,0 +1,165 @@
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.statusCode).toBe(201);
+    expect(res.body).toContain('id');
+    expect(res.body).toContain(testSoldier.id);
+    expect(res.body).toContain(testSoldier.name);
+  });
+
+  it('checks that when soldier not created because of duplicate id status is 400', async () => {
+    await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier2,
+    });
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.statusCode).toBe(400);
+    expect(res.body).toEqual('soldier not created- this id already exists!');
+  });
+});
+
+describe('Get soldier by id route', () => {
+  it('checks that when soldier is found status is 200, and that found soldier is sent', async () => {
+    await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers/${testSoldier.id}`,
+    });
+    expect(res.statusCode).toBe(200);
+    expect(res).toBeTypeOf('object');
+    expect(res.body).toContain(testSoldier.name);
+    expect(res.body).toContain(testSoldier.id);
+  });
+
+  it('checks that when soldier is not found status is 404', async () => {
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers/6${testSoldier.id}`,
+    });
+    expect(res.statusCode).toBe(404);
+  });
+});
+
+describe('Get soldiers route', () => {
+  it('checks that route returns an object', async () => {
+    await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    const res = await app.inject({
+      method: 'GET',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res).toBeTypeOf('object');
+  });
+  it('checks that route returns the relevant object', async () => {
+    await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    const result = await app.inject({
+      method: 'GET',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(result.body).toContain(testSoldier.name);
+  });
+  it('checks that when searched using id it still works', async () => {
+    await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    const res = await app.inject({
+      method: 'GET',
+      url: `/soldiers?name=${testSoldier.name}&id=${testSoldier.id}`,
+    });
+    expect(res).toBeTypeOf('object');
+  });
+});
+
+describe('testing POST/soldiers endpoint', () => {
+  it('checks for status 500 when theres a server error', async () => {
+    const addNewSoldierSpy = vi.spyOn(database, 'addNewSoldier');
+    addNewSoldierSpy.mockImplementation(() => {
+      const err = new Error();
+      err.statusCode = 500;
```

Comment:
In most of the server error, no body will tell you the status code, so just throw an error. also you can add test to check the error message (why?)

Reply:
fixed